### PR TITLE
Add initial parent template

### PR DESCRIPTION
**Pull request by Ricky-Thomas**
### What does this PR do?

Adds initial parents template as well as some tweaks to linked template files.

Run with:

```
New-AzResourceGroupDeployment -TemplateFile $templateFile -resourceGroup "ricky_powershell_rg" -eventHubNamespace "ricky-namespace" -eventHubName "ricky-test-hub" -functionCode $code -apiKey "redacted"  -verbose
```
eventHub and eventHubNamespace are optional arguments here.

### Types of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)
- [ ] This PR passes the unit tests 
- [ ] This PR passes the installation tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: azure/eventhub_log_forwarder/parent_template.json [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Specify a location for the resources."
+      }
+    }
+  },
+  "variables": {
+    "eventHubTemplateLink": "https://raw.githubusercontent.com/DataDog/datadog-serverless-functions/master/azure/eventhub_log_forwarder/event_hub.json",
+    "functionAppTemplateLink": "https://raw.githubusercontent.com/DataDog/datadog-serverless-functions/1db05725d9d1fc78a57968e6f8f95053e1d994e0/azure/eventhub_log_forwarder/function_template.json"
```
Comment (claudiadadamo) [inline azure/eventhub_log_forwarder/parent_template.json]: Should be for the master branch

File: azure/eventhub_log_forwarder/parent_template.json [hunk 2]
```diff
@@ -0,0 +1,101 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
```
Comment (claudiadadamo) [inline azure/eventhub_log_forwarder/parent_template.json:4]: We should have the functionapp and function name in here too right?

File: azure/eventhub_log_forwarder/parent_template.json [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      "name": "functionAppTemplate",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2018-05-01",
+      "properties": {
+        "mode": "Incremental",
+        "templateLink": {
+          "uri": "[variables('functionAppTemplateLink')]",
+          "contentVersion": "1.0.0.0"
+        },
+        "parameters": {
```
Comment (claudiadadamo) [inline azure/eventhub_log_forwarder/parent_template.json:91]: We should pass in the functionapp and function name in here as well

File: azure/eventhub_log_forwarder/parent_template.json [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            "value": "[parameters('location')]"
+          }
+        }
+      },
+      "dependsOn": [
+        "[resourceId('Microsoft.Resources/deployments','eventHubTemplate')]"
+      ]
+    }
+  ],
+  "outputs": {}
```
Comment (claudiadadamo) [inline azure/eventhub_log_forwarder/parent_template.json]: Do we need `outputs` at all?

File: azure/eventhub_log_forwarder/parent_template.json (hunk 1 above)
Comment (Ricky-Thomas) [reply azure/eventhub_log_forwarder/parent_template.json]: Ah yeah, totally forgot to change that. Will update!

File: azure/eventhub_log_forwarder/parent_template.json (hunk 4 above)
Comment (Ricky-Thomas) [reply azure/eventhub_log_forwarder/parent_template.json]: nope haha I cleaned that up, thanks!

File: azure/eventhub_log_forwarder/parent_template.json (hunk 2 above)
Comment (Ricky-Thomas) [reply azure/eventhub_log_forwarder/parent_template.json:4]: Yep, I've added this now!

**Review by claudiadadamo — APPROVED** [review]
lgtm!

**Pull request merged**